bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBridgeHandler.java

```diff
@@ -179,13 +180,49 @@ private void refresh() {
         }
     }
 
+    /**
+     * Check if the data needs to be refreshed.
+     *
+     * The data is refreshed if:
+     * - the thing is offline
+     * - the local cache is empty
+     * - the current time is after 15:00 and the last refresh was more than an hour ago
+     * - the current time is after 18:00 and the last refresh was more than an hour ago
+     * - the current time is after 21:00 and the last refresh was more than an hour ago
+     *
+     * @return true if the data needs to be refreshed
+     */
     private boolean needRefresh() {
+        // if the thing is offline, we need to refresh
         if (getThing().getStatus() != ThingStatus.ONLINE) {
             return true;
         }
-        SortedSet<AwattarPrice> localPrices = prices;
-        return localPrices == null
-                || localPrices.last().timerange().start() < Instant.now().toEpochMilli() + 9 * 3600 * 1000;
+
+        // if the local cache is empty, we need to refresh
+        if (prices != null) {
+            return true;
+        }
+
+        // Note: all this magic is made to avoid refreshing the data too often, since the API is rate-limited
+        // to 100 requests per day.
+
+        // do not refresh before 15:00, since the prices for the next day are available only after 14:00
+        ZonedDateTime now = ZonedDateTime.now(zone);
+        if (now.getHour() < 15) {
```

Comment:
They were often available a few minutes before 14:00 already at the time I wrote the binding, so I would start latest at 14:00 to try to get the new prices.

Reply:
I think 1500 is reasonable as it was before. aWATTar only states that the prices would be available after 1400. So I guess it's ok.

Reply:
I'm going to resolve this for now, feel free to reopen if further discussion is needed.

Reply:
> The later we update the prices, the more it limits the range of bestprice things covering midnight one can create. If we update prices at 14:00 (given that they are available), one can create a bestprice thing from 14:00 until 2:00 next day (for example). If we update prices only after 15:00, such a thing would not work until 15:00.
> 
> Ok, I am not sure whether anybody will ever need this in reality, but only that I can not think of a possible need for this does not mean that it doesn't exist :-)

True, I'll make a follow up, addressing this input. :)